src/extensions/snapshot/snapshot.js

```diff
@@ -0,0 +1,186 @@
+ */
+exports.writeSnapshotFile = (file, content) => {
+    const serializedContent = exports.formatSnapshotFile(content)
+    return fileSystem.writeFileContent(file, serializedContent)
+}
+
+/**
+ * Get snapshot file path base on feature file path
+ * @param {string} featureFile - Feature file path
+ * @param {Object} opts
+ * @param {Object} [opts.snaphotsDirname = '__snapshots__'] - Snapshots dirname
+ * @param {Object} [opts.snapshotsFileExtension = 'snap'] - Snapshots files extension
+ */
+exports.snapshotsPath = (featureFile, opts) => {
+    const dirname = opts.snaphotsDirname || '__snapshots__'
+    const dir = path.join(path.dirname(featureFile), dirname)
+    const filename = `${path.basename(featureFile)}.${opts.snapshotsFileExtension || 'snap'}`
+
+    return path.join(dir, filename)
+}
+
+/**
+ * Compute diff between two contents.
+ * If no diff, it returns null
+ * @param {string} snapshot - snapshot content
+ * @param {string} expected - expected content
+ * @returns {string} Diff message
+ */
+exports.diff = (snapshot, expected) => {
+    let diffMessage = diff(snapshot, expected, {
+        expand: false,
+        colors: true,
+        //contextLines: -1, // Forces to use default from Jest
+        aAnnotation: 'Snapshot',
+        bAnnotation: 'Received'
+    })
+
+    diffMessage =
+        diffMessage ||
+        `${EXPECTED_COLOR('- ' + (expected || ''))} \n ${RECEIVED_COLOR('+ ' + snapshot)}`
+    if (diffMessage === jestDiffConstants.NO_DIFF_MESSAGE) return null
+    return `\n${diffMessage}`
+}
+
+/**
+ * Add backticks to wrap snapshot content and replace backticks
+ * @param {string} str - snapshot content
+ * @return {string} wrapped content
+ */
+exports.printBacktickString = str => {
+    return '`' + str.replace(/`|\\|\${/g, '\\$&') + '`'
+}
+
+/**
+ * Normalize new lines to be \n only
+ * @param {string} string - Content to normalize
+ */
+exports.normalizeNewlines = string => {
+    return string.replace(/\r\n|\r/g, '\n')
+}
+
+/**
+ * For a snapshot file by add backticks and format it as js files with keys
+ * @param {object} content - snapshots content
+ * @return {string} formated snapshot file
+ */
+exports.formatSnapshotFile = content => {
+    const snapshots = Object.keys(content)
+        .sort(naturalCompare)
+        .map(
+            key =>
+                'exports[' +
+                exports.printBacktickString(key) +
+                '] = ' +
+                exports.printBacktickString(exports.normalizeNewlines(content[key])) +
+                ';'
+        )
+    return '\n\n' + snapshots.join('\n\n') + '\n'
+}
+
+/**
+ * Extract keys / values from snapshot file
+ * @param {string} content - Snapshot file content
+ * @return {Object} - should follow the pattern {snapshot_name: snapshot_content}
+ */
+exports.parseSnapshotFile = content => {
+    const data = {}
+    const populate = new Function('exports', content)
```

Comment:
It's really unsafe as you could have a Q&A team writing features and another one implementing those features, having to worry about the security of the snapshots can be cumbersome.

Reply:
Storing it as static data structure, json/yaml…, but it's not such a big deal